packages/asset-swapper/src/utils/market_operation_utils/create_order.ts

```diff
@@ -0,0 +1,168 @@
+            } else {
+                orders.push(
+                    createBridgeOrder(
+                        orderDomain,
+                        this._getBridgeAddressFromSource(source),
+                        outputToken,
+                        inputToken,
+                        fill.output,
+                        fill.input,
+                        bridgeSlippage,
+                    ),
+                );
+            }
+        }
+        return orders;
+    }
+
+    // Convert buy fills into orders.
+    public createBuyOrdersFromPath(
+        orderDomain: OrderDomain,
+        inputToken: string,
+        outputToken: string,
+        path: Fill[],
+        bridgeSlippage: number,
+    ): SignedOrderWithFillableAmounts[] {
+        const orders: SignedOrderWithFillableAmounts[] = [];
+        for (const fill of path) {
+            const source = (fill.fillData as FillData).source;
+            if (source === ERC20BridgeSource.Native) {
+                orders.push((fill.fillData as NativeFillData).order);
+            } else {
+                orders.push(
+                    createBridgeOrder(
+                        orderDomain,
+                        this._getBridgeAddressFromSource(source),
+                        inputToken,
+                        outputToken,
+                        fill.input,
+                        fill.output,
+                        bridgeSlippage,
+                        true,
+                    ),
+                );
+            }
+        }
+        return orders;
+    }
+
+    private _getBridgeAddressFromSource(source: ERC20BridgeSource): string {
+        switch (source) {
+            case ERC20BridgeSource.Eth2Dai:
+                return this._contractAddress.eth2DaiBridge;
+            case ERC20BridgeSource.Kyber:
+                return this._contractAddress.kyberBridge;
+            case ERC20BridgeSource.Uniswap:
+                return this._contractAddress.uniswapBridge;
+            default:
+                break;
+        }
+        throw new Error(AggregationError.NoBridgeForSource);
+    }
+}
+
+function createBridgeOrder(
+    orderDomain: OrderDomain,
+    bridgeAddress: string,
+    makerToken: string,
+    takerToken: string,
+    makerAssetAmount: BigNumber,
+    takerAssetAmount: BigNumber,
+    slippage: number,
+    isBuy: boolean = false,
+): SignedOrderWithFillableAmounts {
+    return {
+        makerAddress: bridgeAddress,
+        makerAssetData: createBridgeAssetData(makerToken, bridgeAddress, createBridgeData(takerToken)),
+        takerAssetData: assetDataUtils.encodeERC20AssetData(takerToken),
+        ...createCommonOrderFields(orderDomain, makerAssetAmount, takerAssetAmount, slippage, isBuy),
+    };
+}
+
+/**
+ * Create ERC20Bridge asset data.
+ */
+export function createBridgeAssetData(tokenAddress: string, bridgeAddress: string, bridgeData: string): string {
+    const encoder = AbiEncoder.createMethod('ERC20Bridge', [
```

Comment:
TODO once we get bridge stuff in `assetDataUtils`.

Reply:
It's already in there :)